lib/src/kvs/ds.rs

```diff
@@ -530,44 +538,65 @@ impl Datastore {
 		Ok(())
 	}
 
-	pub async fn nuke_whole_cluster(&self, tx: &mut Transaction) -> Result<(), Error> {
+	pub async fn clear_unreachable_state(&self, tx: &mut Transaction) -> Result<(), Error> {
 		// Scan nodes
-		let cls = tx.scan_cl(1000).await?;
-		trace!("Found {} nodes", cls.len());
-		for cl in cls {
-			tx.del_cl(
+		let cluster = tx.scan_nd(NO_LIMIT).await?;
+		trace!("Found {} nodes", cluster.len());
+		let mut unreachable_nodes = BTreeMap::new();
+		for cl in &cluster {
+			unreachable_nodes.insert(cl.name.clone(), cl.clone());
+		}
+		// Scan heartbeats
+		let now = tx.clock();
+		let hbs = tx.scan_hb(&now, NO_LIMIT).await?;
```

Comment:
We don't want a timestamp because then unreachable is after now.
Solution scan all (timestamp in future & key entry invalid)
Then correlate and make decision on valid vs invalid.

Reply:
Correction - this is removing invalid nodes aka those without a heartbeat.
Timestamp unnecessary for checks.

Reply:
Change ts to u64 MAX